pkg/types/chain_writer.go

```diff
@@ -27,6 +27,10 @@ type ChainWriter interface {
 type TxMeta struct {
 	// Used for Keystone Workflows
 	WorkflowExecutionID *string
+	// An optional maximum gas limit for the transaction.  It is at the discretion of the chainwriter implementation as
+	// to whether or not to respect this value.  If not set, the chainwriter implementation will be responsible for
+	// setting a gas limit for the transaction.
```

Comment:
Why is it OK/better to ignore a non-nil, as opposed to returning an error?

Reply:
Good question, I suppose if the chainwriter expects this value it could error rather than default when not set, though I guess this would be a breaking change then.   @cedric-cordenier @DeividasK any opinion on this?

Reply:
I mean that in the scenario when a caller sets a `gasLimit`, but the implementation does not support it, then it must either:
1) ignore it
2) return an error

IMHO (2) is safer and more flexible, because it leaves it up to the caller to decide whether dropping the value is acceptable or not. They can choose to submit the request against with a nil value, or to propagate the returned error if having a custom `gasLimit` is required for that situation.

Reply:
I think we'd like to ideally not break CCIP's integration with this change. Considering gasLimit wasn't specified in the TxMeta prevously and was done via the ChainWriterConfig, should we consider supporting that pattern - as in, if the gasLimit isn't specified in the TxMeta, we default to the config value?

Reply:
I've updated the comment and added a new error type as per discussion.

So expected behaviour is that if `gasLimit` is specified and the ChainWriter implementation does not support per transaction gasLimit it will return a `ErrSettingTransactionGasLimitNotSupported` error, this allows the caller to gracefully handle this scenario as per Jordan's suggestion.  In the case where it is not set the ChainWriter implementation will do what it does right now, i.e. default it, so this change will not be breaking (for CCIP).

I'm going to update the evm chainwriter implementation to match the expected behaviour as part of the item fix.  Implementers of other chain writers will need notifying of this change, @patrick-dowell is that anyone other than you?